driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedTestModifications.java

```diff
@@ -372,32 +366,24 @@ private TestApplicator(
                 final TestDef testDef,
                 final List<Modifier> modifiersToApply) {
             this.testDef = testDef;
-            this.shouldSkip = false;
-            this.reasonToApply = null;
             this.modifiersToApply = modifiersToApply;
         }
 
         private TestApplicator(
                 final TestDef testDef,
                 @NonNull
-                final String reason) {
+                final String skipReason) {
```

Comment:
I don't think `skipReason` is used?

Reply:
It is present to force developers to provide the reason (it is not shown in evergreen or I think elsewhere, if used in a failed assumption). I added an `assertNotNull`.